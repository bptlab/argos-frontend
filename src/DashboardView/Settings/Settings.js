import React, { Component } from 'react';
import Modal from '../../Utils/Modal/Modal.js';
import EventTypesSettings from './EventTypes/EventTypes.js';

class Settings extends Component {

    constructor(props) {
        super(props);
        this.state = {
            settingsVisible: false,
        };
        //function binding
        this.modalShow = this.modalShow.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
    }
    
    modalShow() {
        //do not prevent default!
        this.setState({settingsVisible: true});
    }
    
    onSubmit() {
        $('#settings-modal').modal('hide');
        this.setState({settingsVisible: false});
    }
    
    componentDidMount() {
        $('#settings-modal').on('hidden.bs.modal', function () {
            /* istanbul ignore next */
            this.setState({settingsVisible: false});
        }.bind(this));
    }

    render() {
        return (
            <div className="settings">
                <a className="nav-link" data-toggle="modal" data-target="#settings-modal" onClick={this.modalShow}>
                    <i className="settings-icon fa fa-bars" />
                </a>
                <Modal title="Settings" onSubmit={this.onSubmit} id="settings" buttonText="Exit">
                    <div id="accordion" role="tablist" aria-multiselectable="true">
                        <div className="card">
                            <div className="card-header" role="tab" id="headingOne">
                                <h5 className="mb-0">
                                    <a data-toggle="collapse" data-parent="#accordion" href="#collapseOne" 
                                       aria-expanded="true" aria-controls="collapseOne">
                                        Event-Types
                                    </a>
                                </h5>
                            </div>
                            <div id="collapseOne" className="collapse show" role="tabpanel" 
                                 aria-labelledby="headingOne">
                                <div className="card-block">
                                    <EventTypesSettings 
                                        dataSource={this.props.dataSource}
                                        dataSender={this.props.dataSender}
                                        settingsVisible={this.state.settingsVisible} />
                                </div>
                            </div>
                        </div>
                    </div>
                </Modal>
            </div>
        );
    }
}
export default Settings;
