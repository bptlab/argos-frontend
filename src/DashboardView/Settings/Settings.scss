@import "EventTypes/EventTypes";
.card-header h5 a {
  color: $COLOR-font-dark !important;
}

header .settings a, header .settings p, header .settings a:hover, header .settings h1, header .settings {
  color: $COLOR-font-dark !important;
}

.settings-icon {
  color: $COLOR-font-light;
  font-size: 35px;
  @include media-breakpoint-up(uhd) {  
    font-size: 80px;
  }
  cursor: pointer;
}