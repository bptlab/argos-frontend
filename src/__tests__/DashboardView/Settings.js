import React from 'react';
import Settings from './../../DashboardView/Settings/Settings.js';
import renderer from 'react-test-renderer';
let instance, dataSource, dataSender, component;
import 'bootstrap';

beforeEach(() => {
    const notificationService = {
        subscribe: jest.fn(),
        unsubscribe: jest.fn()
    };
    dataSource = {
        fetchAllEventTypes: jest.fn(),
        notificationService: notificationService
    };
    dataSender = {
        notificationService: notificationService
    };
    component = renderer.create(
        <Settings
            dataSource={dataSource}
            ref={(child) => {instance = child}}
            dataSender={dataSender} />
    );
});

test('Correct rendering of Settings', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Correct rendering of Settings after modal has been opened', () => {
    instance.modalShow();
    expect(dataSource.fetchAllEventTypes).toHaveBeenCalled();
    expect(dataSource.notificationService.subscribe).toHaveBeenCalled();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Correct rendering of Settings after modal has been closed', () => {
    instance.modalShow();
    instance.onSubmit();
    expect(dataSource.notificationService.unsubscribe).toHaveBeenCalled();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

