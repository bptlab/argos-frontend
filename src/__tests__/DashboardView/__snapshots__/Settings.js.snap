exports[`test Correct rendering of Settings 1`] = `
<div
  className="settings">
  <a
    className="nav-link"
    data-target="#settings-modal"
    data-toggle="modal"
    onClick={[Function]}>
    <i
      className="settings-icon fa fa-bars" />
  </a>
  <div
    className="settings modal fade"
    id="settings-modal"
    role="dialog"
    tabIndex="-1">
    <div
      className="modal-dialog modal-lg"
      role="document">
      <div
        className="modal-content">
        <div
          className="modal-header">
          <h5
            className="modal-title"
            id="settings-label">
            Settings
          </h5>
          <button
            className="close"
            data-dismiss="modal"
            type="button">
            <span>
              ×
            </span>
          </button>
        </div>
        <div
          className="modal-body">
          <div
            aria-multiselectable="true"
            id="accordion"
            role="tablist">
            <div
              className="card">
              <div
                className="card-header"
                id="headingOne"
                role="tab">
                <h5
                  className="mb-0">
                  <a
                    aria-controls="collapseOne"
                    aria-expanded="true"
                    data-parent="#accordion"
                    data-toggle="collapse"
                    href="#collapseOne">
                    Event-Types
                  </a>
                </h5>
              </div>
              <div
                aria-labelledby="headingOne"
                className="collapse show"
                id="collapseOne"
                role="tabpanel">
                <div
                  className="card-block">
                  <div
                    className="event-types-settings">
                    
                    <table
                      className="table table-striped event-types-table">
                      <thead>
                        <tr>
                          <th>
                            #
                          </th>
                          <th>
                            Name
                          </th>
                          <th>
                            Query
                          </th>
                          <th
                            className="center">
                            Delete?
                          </th>
                        </tr>
                      </thead>
                      <tbody />
                    </table>
                    <p>
                      To add a new Event-Type please refer to a Product and press the \"Plus\"-Button.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="modal-footer">
          <button
            className="btn btn-primary"
            disabled="disabled"
            onClick={[Function]}
            type="button">
            <div>
              Exit
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test Correct rendering of Settings after modal has been closed 1`] = `
<div
  className="settings">
  <a
    className="nav-link"
    data-target="#settings-modal"
    data-toggle="modal"
    onClick={[Function]}>
    <i
      className="settings-icon fa fa-bars" />
  </a>
  <div
    className="settings modal fade"
    id="settings-modal"
    role="dialog"
    tabIndex="-1">
    <div
      className="modal-dialog modal-lg"
      role="document">
      <div
        className="modal-content">
        <div
          className="modal-header">
          <h5
            className="modal-title"
            id="settings-label">
            Settings
          </h5>
          <button
            className="close"
            data-dismiss="modal"
            type="button">
            <span>
              ×
            </span>
          </button>
        </div>
        <div
          className="modal-body">
          <div
            aria-multiselectable="true"
            id="accordion"
            role="tablist">
            <div
              className="card">
              <div
                className="card-header"
                id="headingOne"
                role="tab">
                <h5
                  className="mb-0">
                  <a
                    aria-controls="collapseOne"
                    aria-expanded="true"
                    data-parent="#accordion"
                    data-toggle="collapse"
                    href="#collapseOne">
                    Event-Types
                  </a>
                </h5>
              </div>
              <div
                aria-labelledby="headingOne"
                className="collapse show"
                id="collapseOne"
                role="tabpanel">
                <div
                  className="card-block">
                  <div
                    className="event-types-settings">
                    
                    <table
                      className="table table-striped event-types-table">
                      <thead>
                        <tr>
                          <th>
                            #
                          </th>
                          <th>
                            Name
                          </th>
                          <th>
                            Query
                          </th>
                          <th
                            className="center">
                            Delete?
                          </th>
                        </tr>
                      </thead>
                      <tbody />
                    </table>
                    <p>
                      To add a new Event-Type please refer to a Product and press the \"Plus\"-Button.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="modal-footer">
          <button
            className="btn btn-primary"
            disabled="disabled"
            onClick={[Function]}
            type="button">
            <div>
              Exit
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test Correct rendering of Settings after modal has been opened 1`] = `
<div
  className="settings">
  <a
    className="nav-link"
    data-target="#settings-modal"
    data-toggle="modal"
    onClick={[Function]}>
    <i
      className="settings-icon fa fa-bars" />
  </a>
  <div
    className="settings modal fade"
    id="settings-modal"
    role="dialog"
    tabIndex="-1">
    <div
      className="modal-dialog modal-lg"
      role="document">
      <div
        className="modal-content">
        <div
          className="modal-header">
          <h5
            className="modal-title"
            id="settings-label">
            Settings
          </h5>
          <button
            className="close"
            data-dismiss="modal"
            type="button">
            <span>
              ×
            </span>
          </button>
        </div>
        <div
          className="modal-body">
          <div
            aria-multiselectable="true"
            id="accordion"
            role="tablist">
            <div
              className="card">
              <div
                className="card-header"
                id="headingOne"
                role="tab">
                <h5
                  className="mb-0">
                  <a
                    aria-controls="collapseOne"
                    aria-expanded="true"
                    data-parent="#accordion"
                    data-toggle="collapse"
                    href="#collapseOne">
                    Event-Types
                  </a>
                </h5>
              </div>
              <div
                aria-labelledby="headingOne"
                className="collapse show"
                id="collapseOne"
                role="tabpanel">
                <div
                  className="card-block">
                  <div
                    className="event-types-settings">
                    
                    <table
                      className="table table-striped event-types-table">
                      <thead>
                        <tr>
                          <th>
                            #
                          </th>
                          <th>
                            Name
                          </th>
                          <th>
                            Query
                          </th>
                          <th
                            className="center">
                            Delete?
                          </th>
                        </tr>
                      </thead>
                      <tbody />
                    </table>
                    <p>
                      To add a new Event-Type please refer to a Product and press the \"Plus\"-Button.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="modal-footer">
          <button
            className="btn btn-primary"
            disabled="disabled"
            onClick={[Function]}
            type="button">
            <div>
              Exit
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
